src/test/shell/bazel/bazel_cc_code_coverage_test.sh

```diff
@@ -143,46 +165,68 @@ function run_coverage() {
 
 function test_cc_test_coverage() {
   run_coverage > "$TEST_log"
+  assert_coverage_result
+}
 
-  # After running the test in coverage_srcs/t.cc, the sources covered are the
-  # test itself and the source file a.cc.
-  # For more details about the lcov format see
-  # http://ltp.sourceforge.net/coverage/lcov/geninfo.1.php
-  # The expected result can be constructed manually by following the lcov
-  # documentation and manually checking what lines of code are covered when
-  # running the test.
-  cat <<EOF > expected_result.dat
-TN:
-SF:coverage_srcs/a.cc
-FN:3,_Z1ab
-FNDA:1,_Z1ab
-FNF:1
-FNH:1
-DA:3,1
-DA:4,1
-DA:5,1
-DA:7,0
-LF:4
-LH:3
-end_of_record
-TN:
-SF:coverage_srcs/t.cc
-FN:4,main
-FNDA:1,main
-FNF:1
-FNH:1
-DA:4,1
-DA:5,1
-DA:6,1
-LF:3
-LH:3
-end_of_record
-EOF
-
-  # tools/test/collect_cc_coverage.sh places the coverage result in
-  # $COVERAGE_OUTPUT_FILE
-  diff -u expected_result.dat "$COVERAGE_OUTPUT_FILE_VAR" >> "$TEST_log" \
-    || fail "Coverage output file is different than the expected file"
+# Asserts that the C++ coverage output is correct.
+#
+# The C++ coverage script places the final coverage result in
+# $COVERAGE_OUTPUT_FILE.
+#
+# The covered files are coverage_srcs/a.cc and coverage_srcs/t.cc.
```

Comment:
The cc files should be parameters of this function.

Reply:
I don't think they should. Everything this method asserts is hardcoded and strictly linked to these 2 cc files and nothing else. It doesn't make much sense to pass them as parameters and then have 2 variables corresponding to each of the files hardcoding their relative paths and coverage results.

Reply:
The function is called "assert_coverage_result" which, from the name, I understood it's meant to be general. If that's not the case and there's only one caller then inline it. If you ever need a second test case you'd probably need to generalize it but it's ok for now.

Reply:
It's not meant to be general, it's meant to be the only coverage assertion that should be done when running this test suite.
The purpose of this test suite is to treat the C++ coverage script as a black box and test it for each different coverage tool it supports to make sure they are consistent (and correct ofc). It is important that each coverage tool has the same setup and coverage assertion. The tests should look almost the same, the only thing changing should be a parameter saying which coverage tool to use. I don't see the usefulness of generalizing without a clear reason, especially in testing. Also note that this couldn't really be generalized for the reasons I mentioned in my previous comment. 

This is how the current test looks like now:
```
run_coverage > "$TEST_log"
assert_coverage_result
```

I like it because it is very simple. In the future the same test will be:
```
run_coverage lcov > "$TEST_log"
assert_coverage_result
```

and the following one:
```
run_coverage gcov > "$TEST_log"
assert_coverage_result
```

The downside of course is if/when there is a need for a more complex C++ setup. In that case the setups/assertions can be renamed and grouped together. For the time being I would like to go for the simple solution, especially since it's well documented. If you think I could improve the documentation I will gladly do so.